### PR TITLE
Vote support

### PR DESCRIPTION
**Pull request by PostsDesert**
@stevenleeg  Does this work for you?

### CODE REVIEW

File: geemusic/utils/music.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def index_library(self):
         tracks = self.get_all_songs()
 
         for track in tracks:
-            track, song_id = self.extract_track_info(track)
+            song_id = track['id']
```
Comment (stevenleeg) [inline geemusic/utils/music.py]: ooh, that's what I got wrong

File: geemusic/__init__.py [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 
 app = Flask(__name__)
 ask = Ask(app, '/alexa')
 
-queue = MusicQueue()
 api = GMusicWrapper.generate_api(logger=app.logger)
 
+from utils.music_queue import MusicQueue
+
+queue = MusicQueue()
```
Comment (stevenleeg) [inline geemusic/__init__.py]: is there a reason for moving this and the import down here?

File: geemusic/utils/music.py [hunk 3]
```diff
@@ -108,19 +108,17 @@ def get_all_songs(self):
     def rate_song(self, song, rating):
         return self._api.rate_songs(song, rating)
 
-    @classmethod
-    def extract_track_info(cls, track):
+    # @classmethod
```
Comment (stevenleeg) [inline geemusic/utils/music.py]: could you remove this commented line

File: geemusic/__init__.py (hunk 2 above)
Comment (PostsDesert) [inline geemusic/__init__.py]: Yep, it allows it to import the instance of GMusicWrapper without an error from it trying to load QueueManager before QueueManager is loaded.

File: geemusic/utils/music.py (hunk 3 above)
Comment (PostsDesert) [inline geemusic/utils/music.py]: sure!

**Comment by PostsDesert** [comment]
Thanks for the help refactoring this! It looks much better now.

**Comment by stevenleeg** [comment]
awesome, thanks for working with me on this!

**Comment by stevenleeg** [comment]
also note that I made a slight tweak to remove that import ordering issue: 862b958075f21882076bdfea89ab6c0f7ae5fdf7

**Comment by PostsDesert** [comment]
Looks great! 👍

**Pull request merged**